uvicorn/supervisors/multiprocess.py

```diff
@@ -42,7 +43,9 @@ def signal_handler(self, sig: int, frame: Optional[FrameType]) -> None:
 
     def run(self) -> None:
         self.startup()
-        self.should_exit.wait()
+        while True:
+            if self.should_exit.wait(self.config.reload_delay):
+                break
```

Comment:
This is a setting that should be used only for `--reload`.

Reply:
Oh right, do you think it makes sense to add a config just for this? Or is it better to use a fixed timeout/constant value?

Reply:
I tried to remove the signal added in #1909 to check but the issue was still there on windows. I'll try to diff from 0.21.1 and isolate the features to find the exact change that caused the issue.

Side note: I also noticed also a flaky startup issue happening on windows while debugging this that I'll dig more on it later.

Reply:
Yeah, tests on reload are flaky... I don't know why. You can add the `skipif` on them as well for the time being. I'll accept a PR for it.